### ISSUE TITLE
editor::SelectAllMatches cursor is inconsistent with origin match

### ISSUE THREAD

**Issue by andrewbanchich** [zed-industries/zed#19569]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

in the multi-buffer, if i see

```
foo bar baz
foo bar baz
foo bar baz
```

and select the first bar from right to left so the cursor is here

```
foo |bar baz
foo bar baz
foo bar baz
```

then choose `editor::SelectAllMatches`, the other cursors appear here

```
foo |bar baz
foo bar| baz
foo bar| baz
```

i'd expect them to appear here

```
foo |bar baz
foo |bar baz
foo |bar baz
```

### Environment

Zed: v0.157.5 (Zed)
OS: Linux Wayland ubuntu 24.04
Memory: 31 GiB
Architecture: x86_64
GPU: Intel(R) Graphics (RPL-U) || Intel open-source Mesa driver || Mesa 24.0.9-0ubuntu0.1

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
I am not able to reproduce. I assume this isn't because the issue doesn't exist, but because I'm doing something different than you are.  Could you provide super-detailed step by step instructions (e.g. create a new file with contents x, open find in all file search for string X, press enter, etc) so I can isolate the issue?

**Comment by andrewbanchich** [comment]
[Screencast from 2024-10-29 21-09-46.webm](https://github.com/user-attachments/assets/97ac6867-739c-46da-92a8-d5d3c88c7cba)

**Comment by quintia** [comment]
Isn't this about the behavior being different depending on whether the initial selection is made by moving the cursor from left to right or from right to left?
I think that even if the initial selection is made from right to left, the cursor should move to the right after using `editor::SelectAllMatches`.


https://github.com/user-attachments/assets/3f2c9d04-a4b9-4d65-b520-2b2cc383c763

**Comment by andrewbanchich** [comment]
Yes, the main issue is the initial selection places the cursor where I tell it to based on whether it was right to left or left to right, but the other selections are always "right".

With the current behavior, there isn't a ton I can do with it if the cursor of one of the selections is at the beginning but at the end for everywhere else.

**Comment by andrewbanchich** [comment]
Ideally I'd expect all cursors to be at the same place, and in sync with the initial cursor. So right to left produces all cursors at left, left to right is all cursors at right.

**Comment by quintia** [comment]
That sounds like a good behavior as well. In any case, I also strongly hope for a fix that ensures the cursor positions remain consistent.

**Comment by JosephTLyons** [comment]
Thanks for the report @andrewbanchich - I opened a PR for this.
